feature: add assistant mentions configuration

lenohard opened the pull request:
add configuration of default mentions for Assistant so user don't have to pick the models every time cherry-studio is restarted or mentions are deleted it in another assistant.  It's useful for users who frequently query multiple llm to evaluate them.

function 1: UI 
<img width="1546" height="700" alt="CleanShot 2025-10-14 at 20 33 48@2x" src="https://github.com/user-attachments/assets/38329194-ee7b-41e7-a2a1-4f9faa4f3a03" />

behavior 1:   after configuration, when user select this assistant, mentions show up by default. 
behavior 2:  user can delete or add mentions in inputBar, but for a new topic the default models group would restore.
behavior 3:  user can conveniently save the current selection into assistant configuration by click a button in mention popup panel as follow: 
<img width="824" height="588" alt="CleanShot 2025-10-14 at 20 41 48@2x" src="https://github.com/user-attachments/assets/d8f13b40-2286-4c40-8e13-732fd82e65a7" />

The inputTool in InputBar @'s function is changed. It reflects the current setting of the default mentions's toggle status for a new topic, and users can toggle it to disable or enable the default mentions using cmd+click. The state will be populated to assistant settings.

Commit:
✨ feat: support assistant default model groups

Commit:
✨ feat: enhance assistant default models with quick save and topic reset

- Add "Save as Default" button in mention models quick panel
- Clear mentioned models cache when starting new topic
- Reset to assistant defaults on new topic creation
- Update i18n keys for new save functionality across all locales

Commit:
🐛 fix: improve assistant default models feature

- Add error handling with try-catch for save action
- Use LoggerService instead of console.error
- Remove duplicate model prevention to allow duplicates
- Add save_failed i18n key across all locales
- Sync i18n files for proper key ordering

Commit:
🐛 fix: prevent duplicate default assistant models

EurFelux commented on src/renderer/src/store/migrate.ts:
```diff
@@ -1980,6 +1980,7 @@ const migrateConfig = {
       })
       state.assistants.assistants.forEach((assistant) => {
         updateModelTextDelta(assistant.defaultModel)
+        assistant.defaultModels?.forEach((model) => updateModelTextDelta(model))
```
Do not modify old migration please. Add a new migration if needed.

EurFelux reviewed (changes requested):
By default, enabling multi-model mentions might be a bit annoying, so I suggest adding a toggle switch to control this behavior.

Also, as Dejeune mentioned, we won't consider merging this feature until after the v2 release, as we're currently undertaking a large-scale data architecture overhaul.

lenohard commented on .github/workflows/nightly-build.yml:
sorry, the latest commits are mistakenly updated into this PR branch. I use that to build the package. I would revert them in this branch.

Commit:
🐛 fix: add migration 163 to update defaultModels text delta support

Add a new migration to ensure defaultModels array has proper text delta
support flags set during state hydration. This migration runs after
migration 124 to handle the defaultModels field that was added by the
assistant default models feature.

Commit:
✨ feat: add toggle to enable/disable default model mentions

Add per-assistant control for automatic default model mentions via a toggle switch in assistant settings. When disabled, users must manually select models via @ panel for each new topic instead of auto-populating from defaults.

- Add `enableDefaultModelMentions` field to Assistant type (defaults to true)
- Add inline Switch component in AssistantModelSettings next to "Default Models" label
- Update Inputbar logic to respect toggle setting:
  - Initial state checks toggle before seeding default models
  - Cache restoration respects toggle
  - Default sync respects toggle
  - New topic creation respects toggle when resetting mentions
- Add i18n translations for toggle label across all locales
- Backward compatible: undefined/null defaults to enabled behavior

Files modified:
- src/renderer/src/types/index.ts (line 51)
- src/renderer/src/pages/settings/AssistantSettings/AssistantModelSettings.tsx (lines 42-44, 250-260)
- src/renderer/src/pages/home/Inputbar/Inputbar.tsx (lines 116-118, 232-254, 463-485)
- src/renderer/src/i18n/locales/*.json (enableMentions key)

Commit:
✨ feat: toggle default mentions in inputbar @ button

Transform @ button from opening model picker to per-topic toggle for default mentions.
- State is now per-topic, not persisted to assistant config
- Enabled state (colored) shows default models are active for current conversation
- Disabled state (no color) clears mention chips, user must manually select via @
- Button visual state based on assistant.enableDefaultModelMentions for new topics
- Tooltip shows "enabled/disabled" status with i18n support
- All 9 locale files updated with new i18n keys

Commit:
fix: build errors

Commit:
fix: restore mention toggle behavior
